common/_static/helpfulness.js

```diff
@@ -0,0 +1,345 @@
+			show: function ()
+			{
+				if (!this.closed && !this.completed && this.shouldShow && !this.$container.is(":visible"))
+				{
+					var documentHeight = $(document).height();
+					var windowHeight = $(window).height();
+					var enoughScrollbars = (documentHeight - windowHeight) > 200;
+
+					if ($(window).scrollTop() > 100 || !enoughScrollbars)
+					{
+						this.$container.slideDown('fast');
+					}
+				}
+			},
+
+			submitYes: function()
+			{
+				if (this.targetKeyId)
+				{
+					$(".helpfulness-form").hide();
+					this.showMessage(".processing");
+
+					$.ajax({
+						context: this,
+						url: options.domain + "/base/HelpfulnessLibrary/SubmitYes",
+						type: "POST",
+						data: {
+							pickerId: this.targetPickerId,
+							contentId: this.targetDocumentId,
+							docId: this.targetDocId
+						},
+						success: function (data)
+						{
+							if (data != null && data == "success")
+							{
+								this.showMessage(".success");
+
+								this.setStorage(this.targetKeyId);
+								this.completed = true;
+
+								this.initLoad();
+							}
+							else
+							{
+								this.showMessage(".error");
+							}
+						},
+						error: function (error)
+						{
+							this.showMessage(".error");
+						}
+					});
+				}
+			},
+
+			submitNo: function (value)
+			{
+				if (this.targetKeyId)
+				{
+					$(".helpfulness-form-no").hide();
+					this.showMessage(".processing");
+
+					$.ajax({
+						context: this,
+						url: options.domain + "/base/HelpfulnessLibrary/SubmitNo",
+						type: "POST",
+						data: {
+							pickerId: this.targetPickerId,
+							contentId: this.targetDocumentId,
+							docId: this.targetDocId,
+							reason: value
+						},
+						success: function (data)
+						{
+							if (data != null && data == "success")
+							{
+								this.showMessage(".success");
+
+								this.setStorage(this.targetKeyId);
+								this.completed = true;
+
+								this.initLoad();
+							}
+							else
+							{
+								this.showMessage(".error");
+							}
+						},
+						error: function (error)
+						{
+							this.showMessage(".error");
+						}
+					});
+				}
+			},
+
+			getStorage: function (contentId)
+			{
+				if (window.localStorage)
+				{
+					return localStorage.getItem("helpfulness-" + contentId);
+				}
+
+				return null;
+			},
+
+			setStorage: function (contentId)
+			{
+				if (window.localStorage)
+				{
+					try
+					{
+						localStorage.setItem("helpfulness-" + contentId, 1);
+						return true;
+					}
+					catch (e)
+					{
+						if (e.name === 'QUOTA_EXCEEDED_ERR' || e.name === 'NS_ERROR_DOM_QUOTA_REACHED')
+						{
+							return false;
+						}
+						else
+						{
+							return false;
+						}
+					}
+				}
+
+				return false;
+			},
+
+			showMessage: function (message)
+			{
+				this.$content.children(".messages").hide();
+				this.$content.children(message).show();
+			}
+		};
+	};
+})(jQuery);
+
+$(function ()
+{
+	$.Helpfulness({
+		maxCharacters: 160,
```

Comment:
I think it would be very unfortunate to have only 160 chars. On so many sites with even a 500 char limit I've been frustrated that I can barely type in a sentence and include one URL and I'm out of space.

The scenario that happens to me is this:
1. I'm annoyed at the product I'm using because something doesn't work
2. I find the doc site to find info on how to fix the problem I'm having
3. The doc site is lacking in some way and I didn't find the help I wanted
4. A popup says "do you have any feedback"?
5. I say "yes" and type in my feedback
6. When I'm not even half way done typing, I'm thwarted by the site that wanted my help and doesn't even let me help.
7. Now I'm even more annoyed at the site/docs/product

Please, 1000 chars! :smile: